starlette/_crypto.py

```diff
@@ -0,0 +1,14 @@
+import hashlib
+import inspect
+
+
+# TODO: Remove when dropping support for PY38. inspect.signature() is used to
+# detect whether the usedforsecurity argument is available as this fix may also
+# have been applied by downstream package maintainers to other versions in
+# their repositories.
+if "usedforsecurity" in inspect.signature(hashlib.md5).parameters:
```

Comment:
I decided to just check for the parameter directly here. Django has a bunch of utility functions around this, but we're only doing it once here so that seemed like overkill.

Reply:
Just for my curiosity, is the `usedforsecurity` based exclusively on the Python version? Is there an environment with Python 3.8+ that will not have `usedforsecurity` parameter?

Reply:
I think the reason Django did that is that some enterprises might have backported the fix to some internal fork, or something like that. Figured we might as well follow what they did. But I'm happy to just check `sys.version_info`.

Since this is done once at import time, I don't think performance matters.

Reply:
I don't see how a _backport_ by checking the versions would be different from checking the parameter :thinking:
It looks like both are interchangeable, or am I missing something? Because if I'm not missing anything, then we could avoid the comment and just check the version :thinking:

Reply:
I think the point is that the parameter _could_ be present in Python 3.7 if someone backported it internally. Invoking @vakwetu (author of the Django PR) in case they can clarify.

Reply:
Yes, some distros have in fact backported this parameter.  Red Hat/Fedora/Centos has backported this to python 3.6 IIRC.

Reply:
Awesome, thank you for confirming!